opensrp-app/src/main/java/org/smartregister/util/FormUtils.java

```diff
@@ -1121,4 +1123,38 @@ private void createNewEventDocument(org.smartregister.cloudant.models.Event even
     private void createNewClientDocument(org.smartregister.cloudant.models.Client client) {
         mCloudantDataHandler.createClientDocument(client);
     }
+
+    public JSONObject getFormJsonFromRepositoryOrAssets(String formIdentity) {
+        ClientFormRepository clientFormRepository = CoreLibrary.getInstance().context().getClientFormRepository();
+        String locale = mContext.getResources().getConfiguration().locale.getLanguage();
+
+        //Check the current locale of the app to load the correct version of the form in the desired language
+        String localeFormIdentity = formIdentity;
+        if (!locale.equals("en")) {
+            localeFormIdentity = localeFormIdentity + "-" + locale;
+        }
+
+        ClientForm clientForm = clientFormRepository.getActiveClientFormByIdentifier(localeFormIdentity);
+        try {
+            if (clientForm != null) {
+                Timber.d("============%s form loaded from db============", formIdentity);
+                String jsonString = convertStandardJSONString(clientForm.getJson());
+                return new JSONObject(jsonString);
+            }
+        } catch (JSONException e) {
+            Timber.e(e);
+        }
+        Timber.d("============%s form loaded from Assets=============", formIdentity);
+        return getFormJson(formIdentity);
+    }
+
+    private String convertStandardJSONString(String data_json) {
```

Comment:
I was thinking that this code will be difficult to maintain and  might not cover all the cases. I think this `StringEscapeUtils.unescapeJson(clientFormResponse.getClientForm().getJson())` is better. I also think that we should save the json unescaped just like the event & client JSONs are saved in the tables. This will mean that you only have to unescape it once during download. After that, the JSON will be like standard JSON, easy to ready, debug & work with

![Screenshot from 2020-04-24 11-28-13](https://user-images.githubusercontent.com/31766075/80191765-d93cc200-861e-11ea-9ac5-fc672bb724e5.png)

Reply:
Cool this makes more sense